db/src/db.rs

```diff
@@ -146,7 +146,16 @@ lazy_static! {
         r#"CREATE UNIQUE INDEX idx_datoms_unique_value ON datoms (a, value_type_tag, v) WHERE unique_value IS NOT 0"#,
 
         r#"CREATE TABLE transactions (e INTEGER NOT NULL, a SMALLINT NOT NULL, v BLOB NOT NULL, tx INTEGER NOT NULL, added TINYINT NOT NULL DEFAULT 1, value_type_tag SMALLINT NOT NULL)"#,
-        r#"CREATE INDEX idx_transactions_tx ON transactions (tx, added)"#,
+
+        // This index serves two purposes:
+        //
+        // - First, it allows to query specific transactions efficiently, and then to filter by
+        //   attribute in a transaction efficiently.
+        //
+        // - Second, it provides a layer of assurance that the transactor is correct by asserting
+        //   that a single dato `[e a v]` is added or retracted at most once in a single
+        //   transaction.
+        r#"CREATE UNIQUE INDEX idx_transactions_unique_tx_a_e_v ON transactions (tx, a, e, v)"#,
```

Comment:
Good opportunity to write a disk-level migration as an example, and avoid breaking any test DBs…

Reply:
Meh, I don't really care to do this.  I'm going to leave as is and think more about what indexing we really want post-trie.